.github/workflows/build.yml

```diff
@@ -65,49 +77,62 @@ jobs:
               uses: docker/build-push-action@v5
               with:
                   context: .
-                  platforms: linux/amd64,linux/arm64
-                  push: ${{ github.ref == 'refs/heads/develop' }}
+                  platforms: ${{ matrix.platforms }}
+                  push: ${{ github.ref == 'refs/heads/develop' || true }}
                   file: docker/dockerfile
-                  tags: ${{ needs.set-vars.outputs.CI_IMAGE }}
-                  cache-from: type=registry,ref=${{ needs.set-vars.outputs.CI_IMAGE }}
+                  tags: ${{ needs.set-vars.outputs.CI_IMAGE }}:${{ matrix.backend }}
+                  cache-from: type=registry,ref=${{ needs.set-vars.outputs.CI_IMAGE }}:${{ matrix.backend }}
                   cache-to: type=inline
+                  target: final
                   build-args: |
+                      BASE_IMAGE=${{ matrix.base_image }}
+                      CMAKE_VERSION=${{ env.CMAKE_VERSION }}
                       DOXYGEN_VERSION=${{ env.DOXYGEN_VERSION }}
                       GOOGLETEST_VERSION=${{ env.GOOGLETEST_VERSION }}
+                      KOKKOS_ARCH=${{ matrix.kokkos_arch }}
+                      KOKKOS_PRESET=${{ matrix.backend }}
                       KOKKOS_VERSION=${{ env.KOKKOS_VERSION }}
-                      KOKKOS_PRESET=OpenMP
                   labels: "org.opencontainers.image.source=${{ github.repositoryUrl }}"
 
-    build-library:
+    build-library-and-test:
         needs: [set-vars, build-image]
         runs-on: [ubuntu-latest]
+        strategy:
+            fail-fast: false
+            matrix:
+                include:
+                    - backend: Cuda
+                      tests  : false
```

Comment:
The name for the flag could also be `run_tests` to emphasise that we compile the tests, but it's controlling whether we run the tests.

Reply:
OK, let's go !

Reply:
Done.